src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -48,8 +48,6 @@ public sealed class BlockchainProcessor : IBlockchainProcessor, IBlockProcessing
         {
             // Optimize for single reader concurrency
             SingleReader = true,
-            // Share thread with request, if waiting
-            AllowSynchronousContinuations = true
```

Comment:
Not convinced about this one, I think it should work well in production when we only get single request? @benaadams

Reply:
This one in particular reliably locks `BlockchainProcessorTests` when run on its own. I don't recommend these kind per block optimization, it took a lot of time to trace this kind of problem.

Reply:
Bigger issue is test parameter discovery triggering all the static test parameter values, which then start blockchains, when then setup background pruning tasks etc, which then soak up all the threads (as happening with many tests in parallel), just to decide it needs to put 0x521 in as a parameter to a test case 🤷‍♂️

Reply:
@benaadams I have understood none of that. I will leave it up to you.